generic-server/README.md

```diff
@@ -1,2 +1,51 @@
 # generic-server
-This server aggregates features provided by configurable plugins.
+
+This server provides a system for building versatile applications by integrating functionalities offered by configurable plugins.
+## Features
+
+* **Plugin System:** Enables developers to extend server behavior through custom plugins.
+* **Routing and Transport:** Handles communication between clients and plugins using industry-standard protocols (e.g., HTTP).
+
+# Usage
+## Extending Server Functionalities with Plugins
+### To add a new plugin:
+
+**1- Implement the `Plugin` Trait:**
+
+- Define the necessary methods (e.g., name(), mount(), and routes()).
+
+**2-Register the Plugin:**
+
+- Add the plugin to the static PLUGINS array.
+
+**3-Load the Plugin:**
+
+- Ensure the plugin container is re-initialized to include the new plugin.**
```

Comment:
This step is not necessary, the plugins are loaded automatically as the server boots. I can't recall the plugin container being able to be re-initialized.